tests/std/tests/P1522R1_difference_type/test.cpp

```diff
@@ -0,0 +1,529 @@
+        assert(-product + product == 0);
+
+        product >>= 17;
+        assert(product._Word[0] == 0x85b70639'545b523d);
+        assert(product._Word[1] == 0xffffedaf'390f35b0);
+        assert(+product == product);
+        assert(-product + product == 0);
+    }
+
+    {
+        _Int128 q = _Int128{13} / _Int128{4};
+        assert(q._Word[0] == 3);
+        assert(q._Word[1] == 0);
+
+        q = _Int128{4} / _Int128{13};
+        assert(q._Word[0] == 0);
+        assert(q._Word[1] == 0);
+
+        q = _Int128{0x01010101'01010101, 0x01010101'01010101} / _Int128{13};
+        assert(q._Word[0] == 0xc50013c5'0013c500);
+        assert(q._Word[1] == 0x13c500'13c50013);
+
+        q = _Int128{0x22222222'22222222, 0x22222222'22222222} / _Int128{0x11111111'11111111, 0x11111111'11111111};
+        assert(q._Word[0] == 2);
+        assert(q._Word[1] == 0);
+
+        q = (_Int128{1} << 66) / (_Int128{1} << 13);
+        assert(q._Word[0] == 1ull << 53);
+        assert(q._Word[1] == 0);
+
+        auto tmp = ~_Int128{};
+        assert(tmp._Word[0] == ~0ull);
+        assert(tmp._Word[1] == ~0ull);
+        q = tmp / std::uint32_t{1};
+        assert(q == tmp);
+        q = tmp / std::uint64_t{1};
+        assert(q == tmp);
+        q = tmp / _Int128{1};
+        assert(q == tmp);
+
+        q = tmp / _Int128{1ull << 32};
+        assert(q._Word[0] == 0);
+        assert(q._Word[1] == 0);
+
+        _Int128 result{0x80808080'80808080, 0x808080'80808080};
+        for (tmp = 0xffu; tmp > 0; tmp <<= 8, result >>= 8) {
+            q = std::numeric_limits<_Int128>::max() / tmp;
+            assert(q == result);
+            auto m    = std::numeric_limits<_Int128>::max() % tmp;
+            auto p    = q * tmp;
+            auto diff = std::numeric_limits<_Int128>::max() - p;
+            assert(m == diff);
+        }
+
+        result = _Int128{0x7f7f7f7f'7f7f7f80, 0xff7f7f7f'7f7f7f7f};
+        for (tmp = 0xffu; tmp > 0; tmp <<= 8, result = (result >> 8) + 1) {
+            q = std::numeric_limits<_Int128>::min() / tmp;
+            assert(q == result);
+            auto m    = std::numeric_limits<_Int128>::min() % tmp;
+            auto p    = q * tmp;
+            auto diff = std::numeric_limits<_Int128>::min() - p;
+            assert(m == diff);
+        }
+    }
+
+    {
+        const _Int128 x{0x01020304'02030405, 0x03040506'04050607};
+        const _Int128 y{0x07060504'06050403, 0x05040302'04030101};
+        assert(((x & y) == _Int128{0x01020104'02010401, 0x01040102'04010001}));
+        auto tmp = x;
+        tmp &= y;
+        assert(tmp == (x & y));
+
+        assert(((x | y) == _Int128{0x07060704'06070407, 0x07040706'04070707}));
+        tmp = x;
+        tmp |= y;
+        assert(tmp == (x | y));
+
+        assert(((x ^ y) == _Int128{0x06040600'04060006, 0x06000604'00060706}));
+        tmp = x;
+        tmp ^= y;
+        assert(tmp == (x ^ y));
+    }
+
+    return true;
+}
+
+int main() {

```

Comment:
Apologies if this exists and I missed it - is there test coverage of "usual arithmetic conversions", i.e. signed op unsigned and unsigned op signed, where unsigned wins? Just want to make sure it's not ambiguous.